core/server/src/main/java/alluxio/master/file/async/DefaultAsyncPersistHandler.java

```diff
@@ -0,0 +1,165 @@
+import alluxio.exception.FailedToCheckpointException;
+import alluxio.exception.FileDoesNotExistException;
+import alluxio.exception.InvalidPathException;
+import alluxio.master.file.FileSystemMaster;
+import alluxio.master.file.meta.FileSystemMasterView;
+import alluxio.thrift.PersistFile;
+import alluxio.util.IdUtils;
+import alluxio.wire.BlockLocation;
+import alluxio.wire.FileBlockInfo;
+import alluxio.wire.FileInfo;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The default handler for async persistence that schedules the the persistence on the workers that
+ * contains all the blocks of a given file, and the handler returns the scheduled request whenever
+ * the corresponding worker polls.
+ */
+public class DefaultAsyncPersistHandler implements AsyncPersistHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private final FileSystemMasterView mFileSystemMasterView;
+
+  /** Map from worker to the files to persist on that worker. Used by async persistence service. */
+  private final Map<Long, Set<Long>> mWorkerToAsyncPersistFiles;
+
+  /**
+   * Constructs a new instance of {@link AsyncPersistHandler}.
+   *
+   * @param view a view of {@link FileSystemMaster}
+   */
+  public DefaultAsyncPersistHandler(FileSystemMasterView view) {
+    mWorkerToAsyncPersistFiles = Maps.newHashMap();
+    mFileSystemMasterView = Preconditions.checkNotNull(view);
+  }
+
+  @Override
+  public synchronized void scheduleAsyncPersistence(AlluxioURI path)
+      throws AlluxioException {
+    // find the worker
+    long workerId = getWorkerStoringFile(path);
+
+    if (workerId == IdUtils.INVALID_WORKER_ID) {
+      throw new FailedToCheckpointException(
+          "No worker found to schedule async persistence for file " + path);
+    }
+
+    if (!mWorkerToAsyncPersistFiles.containsKey(workerId)) {
+      mWorkerToAsyncPersistFiles.put(workerId, Sets.<Long>newHashSet());
+    }
+    mWorkerToAsyncPersistFiles.get(workerId).add(mFileSystemMasterView.getFileId(path));
+  }
+
+  /**
+   * Gets a worker where the given file is stored.
+   *
+   * @param path the path to the file
+   * @return the id of the storing worker
+   * @throws FileDoesNotExistException when the file does not exist on any worker
+   * @throws AccessControlException if permission checking fails
+   */
+  // TODO(calvin): Propagate the exceptions in certain cases
+  private long getWorkerStoringFile(AlluxioURI path)
+      throws FileDoesNotExistException, AccessControlException {
+    Map<Long, Integer> workerBlockCounts = Maps.newHashMap();
+    List<FileBlockInfo> blockInfoList;
+    try {
+      blockInfoList = mFileSystemMasterView.getFileBlockInfoList(path);
```

Comment:
@pfxuan Is there any other error message in the log?

Also, what's the size of `test.txt`? This could happen when either the file is not stored in Alluxio, or the blocks are not stored on the same worker.